HARK/ConsumptionSaving/ConsRiskyAssetModel.py

```diff
@@ -239,9 +357,919 @@ def get_shocks(self):
         self.get_Adjust()
 
 
-# %% Initial parameter sets
+# This is to preserve compatibility with old name
+RiskyAssetConsumerType = IndShockRiskyAssetConsumerType
+
+
+class FixedPortfolioShareRiskyAssetConsumerType(IndShockRiskyAssetConsumerType):
+    time_vary_ = IndShockRiskyAssetConsumerType.time_vary_ + ["RiskyShareFixed"]
+
+    def __init__(self, verbose=False, quiet=False, **kwds):
+        params = init_risky_share_fixed.copy()
+        params.update(kwds)
+        kwds = params
+
+        # Initialize a basic consumer type
+        IndShockRiskyAssetConsumerType.__init__(
+            self, verbose=verbose, quiet=quiet, **kwds
+        )
+
+        self.solve_one_period = make_one_period_oo_solver(
+            ConsFixedPortfolioIndShkRiskyAssetSolver
+        )
+
+
+####################################################################################################
+####################################################################################################
+
+
+@dataclass
+class ConsIndShkRiskyAssetSolver(ConsIndShockSolver):
```

Comment:
I don't have a deep insight into this, but superficially, I'm surprised that the three solver classes are each so long.

I would have expected more of an opportunity for code reuse between these solvers, given their similarity.

I wonder whether there's any functional redundancies in these classes that could be factored out.